File: AspNetMvcUnleashed/0672329980_source/Chapter05/VB/EFMvcApplication/EFMvcApplication/Controllers/ProductCountController.vb
Public Class ProductCountController
	Inherits Controller

	Private _repository As IRepository

	Public Sub New()
		_repository = New Repository(New ToyStoreDBEntities())
	End Sub

	'
	' GET: /ProductCount/

	Public Function Index() As Integer
		Return _repository.GetProductCount()
	End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter05/CS/EFMvcApplication/EFMvcApplication/Controllers/ProductCountController.cs
    using System.Web.Mvc;
    using EFMvcApplication.Models;

    namespace EFMvcApplication.Controllers
    {
        public class ProductCountController : Controller
        {
            private IRepository _repository;

            public ProductCountController()
            {
                _repository = new Repository(new ToyStoreDBEntities());
            }

            //
            // GET: /ProductCount/

            public int Index()
            {
                return _repository.GetProductCount();
            }

        }
    }
